site/en/docs/extensions/reference/scripting/index.md

```diff
@@ -2,9 +2,191 @@
+directory. The following code will inject the file `script.js` into the main
+frame of the tab.
+
+```js
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId},
+      files: ['script.js'],
+    },
+    () => { ... });
+```
+
 !!!.aside.warning
-This documentation is in active development.
-We're planning to expand the introduction section to provide more context and guidance on how to use this API.
-This work is tracked in issue [#194](https://github.com/GoogleChrome/developer.chrome.com/issues/194).
+Note: Currently, a maximum of a single file is supported.
 !!!
 
+#### Runtime Functions
+
+When injecting JavaScript with `scripting.executeScript()`, you can specify a
+function to be executed instead of a file. This function should be a function
+variable available to the current extension context.
+
+```js
+function getTitle() {
+  return document.title;
+}
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId},
+      function: getTitle,
+    },
+    () => { ... });
+```
+
+This function will be executed in the context of injection target. However,
+this will not carry over any of the current execution context of the function.
+As such, bound parameters (including the `this` object) and
+externally-referenced variables will result in errors.  For instance, the
+following code will not work, and will throw a ReferenceError because `color`
+is undefined when the function executes:
+
+```js
+const color = getUserColor();
+function changeBackgroundColor() {
+  document.body.style.backgroundColor = color;
+}
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId},
+      function: changeBackgroundColor,
+    },
+    () => { ... });
+```
+
+You can work around this by using the [Storage API](storage) or by
+[passing messages](messaging).
+
+The Chromium team is working on providing more capabilities in the future.
+
+#### Runtime Strings
+
+If injecting CSS within a page, you can also specify a string to be used in the
+`css` property. Note that this option is only available for
+`scripting.insertCSS()`; you cannot execute a string via
+`scripting.executeScript()`.
+
+```js
+const css = 'body { background-color = "red"; }';
+const tabId = getTabId();
+chrome.scripting.insertCSS(
+    {
+      target: {tabId: tabId},
+      css: css,
+    },
+    () => { ... });
+```
+
+### Handling Results
+
+The results of executing JavaScript are passed to the extension. A single
+result is included per-frame. The main frame is guaranteed to be the first
+index in the resulting array; all other frames are in a non-deterministic
+order.
+
+```js
+function getTitle() {
+  return document.title;
+}
+const tabId = getTabId();
+chrome.scripting.executeScript(
+    {
+      target: {tabId: tabId, allFrames: true},
+      function: getTitle,
+    },
+    (injectionResults) => {
+      for (const frameResult : injectionResults)
```

Comment:
I think you've been writing too much C++: `:` should probably be `of`.

Reply:
Guilty!  Fixed.